### ISSUE TITLE
failure on dc_mlx5/uct_p2p_mix_test_alloc_methods.mix1000_odp/0

### ISSUE THREAD

**Issue by alinask** [openucx/ucx#3453]
```
08:56:26 [ RUN      ] dc_mlx5/uct_p2p_mix_test_alloc_methods.mix1000_odp/0
08:56:26 /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../test/gtest/uct/test_p2p_mix.cc:27: Failure
08:56:26 Value of: _status
08:56:26   Actual: -3
08:56:26 Expected: UCS_OK
08:56:26 Which is: 0
08:56:26 Error: Input/output error
08:56:26 [hpc-test-node3:6196 :0:6196] ib_mlx5_log.c:139  Local protection on mlx5_1:1/IB (synd 0x4 vend 0x56 hw_synd 0/0)
08:56:26 [hpc-test-node3:6196 :0:6196] ib_mlx5_log.c:139  DCI QP 0x3ebcb wqe[6]: SEND s-e [rqpn 0xc0 rlid 27] [inl len 59] [va 0x7ffff7e01039 len 199 lkey 0x81a22]
08:56:26 
08:56:26 /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../src/ucs/debug/assert.c: [ ucs_fatal_error_message() ]
08:56:26       ...
08:56:26        33     }
08:56:26        34 
08:56:26        35     ucs_handle_error(message_buf);
08:56:26 ==>    36     abort();
08:56:26        37 }
08:56:26        38 
08:56:26        39 void ucs_fatal_error_format(const char *file, unsigned line,
08:56:26 
08:56:27 ==== backtrace (tid:   6196) ====
08:56:27  0 0x000000000004dbe0 ucs_fatal_error_message()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../src/ucs/debug/assert.c:36
08:56:27  1 0x0000000000052a03 ucs_log_default_handler()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../src/ucs/debug/log.c:140
08:56:27  2 0x0000000000052b3b ucs_log_dispatch()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../src/ucs/debug/log.c:193
08:56:27  3 0x000000000001fcba uct_ib_mlx5_completion_with_err()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../src/uct/ib/mlx5/ib_mlx5_log.c:132
08:56:27  4 0x0000000000063e66 uct_ib_mlx5_poll_cq()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../src/uct/ib/mlx5/ib_mlx5.inl:38
08:56:27  5 0x0000000000620d1a uct_test::progress()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../src/ucs/datastruct/callbackq.h:211
08:56:27  6 0x00000000005eb650 uct_p2p_mix_test::random_op()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../test/gtest/uct/test_p2p_mix.cc:132
08:56:27  7 0x00000000005eba46 uct_p2p_mix_test::run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../test/gtest/uct/test_p2p_mix.cc:156
08:56:27  8 0x000000000054d4e2 ucs::test_base::run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../test/gtest/common/test.cc:262
08:56:27  9 0x000000000054d5e8 ucs::test_base::TestBodyProxy()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../test/gtest/common/test.cc:288
08:56:27 10 0x000000000053d5ca testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../test/gtest/common/gtest-all.cc:3562
08:56:27 11 0x00000000005356b0 testing::Test::Run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../test/gtest/common/gtest-all.cc:3634
08:56:27 12 0x0000000000535a15 testing::Test::Run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../test/gtest/common/gtest-all.cc:3626
08:56:27 13 0x0000000000535b4d testing::TestInfo::Run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../test/gtest/common/gtest-all.cc:3929
08:56:27 14 0x000000000053658d testing::TestCase::Run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../test/gtest/common/gtest-all.cc:5800
08:56:27 15 0x0000000000536805 testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../test/gtest/common/gtest-all.cc:5719
08:56:27 16 0x000000000051e7f6 RUN_ALL_TESTS()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/1/contrib/../test/gtest/common/gtest.h:20059
08:56:27 17 0x0000000000021c05 __libc_start_main()  ???:0
08:56:27 18 0x000000000051f3e8 _start()  ???:0
08:56:27 =================================
```

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9699/label=hpc-test-node3,worker=1/console

fw_ver:                         16.24.1003
MLNX_OFED_LINUX-4.5-1.0.1.0

**Comment by dmitrygladkov** [comment]
This test hangs with RC_MLX5

```
09:54:05 [ RUN      ] rc_mlx5/uct_p2p_mix_test_alloc_methods.mix1000_odp/2
10:09:05 /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/test_helpers.cc:42: Failure
10:09:05 Failed
10:09:05 Connection timed out - abort testing
10:09:05 [hpc-test-node3:34927:0:34927] Caught signal 6 (Aborted: tkill(2) or tgkill(2))
10:09:06 
10:09:06 /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../src/uct/ib/rc/accel/rc_mlx5.inl: [ uct_rc_mlx5_iface_common_poll_rx() ]
10:09:06       ...
10:09:06      1041 #endif
10:09:06      1042 
10:09:06      1043     ucs_assert(uct_ib_mlx5_srq_get_wqe(&iface->rx.srq,
10:09:06 ==>  1044                                        iface->rx.srq.mask)->srq.next_wqe_index == 0);
10:09:06      1045 
10:09:06      1046     cqe = uct_rc_mlx5_iface_poll_rx_cq(iface);
10:09:06      1047     if (cqe == NULL) {
10:09:06 
10:09:06 ==== backtrace (tid:  34927) ====
10:09:06  0 0x0000000000040c2e uct_rc_mlx5_iface_common_poll_rx()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../src/uct/ib/rc/accel/rc_mlx5.inl:1044
10:09:06  1 0x00000000006245ca uct_test::progress()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../src/ucs/datastruct/callbackq.h:211
10:09:06  2 0x00000000005f10c8 uct_p2p_mix_test::random_op()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../test/gtest/uct/test_p2p_mix.cc:132
10:09:06  3 0x00000000005f2680 uct_p2p_mix_test::run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../test/gtest/uct/test_p2p_mix.cc:156
10:09:06  4 0x00000000005586fe ucs::test_base::run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/test.cc:262
10:09:06  5 0x0000000000547a93 testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/gtest-all.cc:3562
10:09:06  6 0x000000000053bf0d testing::Test::Run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/gtest-all.cc:3635
10:09:06  7 0x000000000053bfdc testing::TestInfo::Run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/gtest-all.cc:3812
10:09:06  8 0x000000000053c13f testing::TestCase::Run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/gtest-all.cc:3930
10:09:06  9 0x0000000000540ad7 testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/gtest-all.cc:5802
10:09:06 10 0x0000000000540ddb testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/gtest-all.cc:5719
10:09:06 11 0x00000000004e8753 main()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/gtest.h:20059
10:09:06 12 0x0000000000021c05 __libc_start_main()  ???:0
10:09:06 13 0x000000000052669e _start()  ???:0
10:09:06 =================================
10:09:06 Sending notification to dmitrygla@mellanox.com
10:09:09 [hpc-test-node3:34927:0:34927] Process frozen..
```

**Comment by dmitrygladkov** [comment]
```
20:11:37 [     INFO ] Using random seed of 30025
...
20:35:37 [----------] 2 tests from dc_mlx5/uct_p2p_mix_test_alloc_methods
20:35:37 [ RUN      ] dc_mlx5/uct_p2p_mix_test_alloc_methods.mix1000_odp/2
20:50:37 /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/2/contrib/../test/gtest/common/test_helpers.cc:42: Failure
20:50:37 Failed
20:50:37 Connection timed out - abort testing
20:50:37 [hpc-test-node3:61109:0:61109] Caught signal 6 (Aborted: tkill(2) or tgkill(2))
20:50:38 
20:50:38 /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/2/contrib/../src/uct/ib/mlx5/ib_mlx5.inl: [ uct_ib_mlx5_get_cqe() ]
20:50:38       ...
20:50:38        10 uct_ib_mlx5_get_cqe(uct_ib_mlx5_cq_t *cq,  unsigned index)
20:50:38        11 {
20:50:38        12     return cq->cq_buf + ((index & (cq->cq_length - 1)) << cq->cqe_size_log);
20:50:38 ==>    13 }
20:50:38        14 
20:50:38        15 static UCS_F_ALWAYS_INLINE int
20:50:38        16 uct_ib_mlx5_cqe_is_hw_owned(uint8_t op_own, unsigned index, unsigned mask)
20:50:38 
20:50:38 ==== backtrace (tid:  61109) ====
20:50:38  0 0x0000000000060178 uct_ib_mlx5_get_cqe()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/2/contrib/../src/uct/ib/mlx5/ib_mlx5.inl:13
20:50:38  1 0x0000000000624b5a uct_test::progress()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/2/contrib/../src/ucs/datastruct/callbackq.h:211
20:50:38  2 0x00000000005f1448 uct_p2p_mix_test::random_op()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/2/contrib/../test/gtest/uct/test_p2p_mix.cc:132
20:50:38  3 0x00000000005f2a00 uct_p2p_mix_test::run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/2/contrib/../test/gtest/uct/test_p2p_mix.cc:156
20:50:38  4 0x0000000000557f3e ucs::test_base::run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/2/contrib/../test/gtest/common/test.cc:273
20:50:38  5 0x00000000005472d3 testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/2/contrib/../test/gtest/common/gtest-all.cc:3562
20:50:38  6 0x000000000053b74d testing::Test::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/2/contrib/../test/gtest/common/gtest-all.cc:3635
20:50:38  7 0x000000000053b81c testing::TestInfo::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/2/contrib/../test/gtest/common/gtest-all.cc:3812
20:50:38  8 0x000000000053b97f testing::TestCase::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/2/contrib/../test/gtest/common/gtest-all.cc:3930
20:50:38  9 0x0000000000540317 testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/2/contrib/../test/gtest/common/gtest-all.cc:5802
20:50:38 10 0x000000000054061b testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/2/contrib/../test/gtest/common/gtest-all.cc:5719
20:50:38 11 0x00000000004e7f43 main()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/2/contrib/../test/gtest/common/gtest.h:20059
20:50:38 12 0x0000000000021c05 __libc_start_main()  ???:0
20:50:38 13 0x0000000000525ede _start()  ???:0
20:50:38 =================================
20:50:38 Sending notification to dmitrygla@mellanox.com
20:50:40 [hpc-test-node3:61109:0:61109] Process frozen...
```

**Comment by dmitrygladkov** [comment]
```
08:49:51 [ RUN      ] rc_mlx5/uct_p2p_mix_test_alloc_methods.mix1000_odp/1
09:04:51 /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/0/contrib/../test/gtest/common/test_helpers.cc:46: Failure
09:04:51 Failed
09:04:51 Connection timed out - abort testing
09:04:51 [hpc-test-node3:39271:0:39271] Caught signal 6 (Aborted: tkill(2) or tgkill(2))
09:04:52 
09:04:52 /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/0/contrib/../src/uct/ib/rc/accel/rc_mlx5.inl: [ uct_rc_mlx5_iface_common_poll_rx() ]
09:04:52       ...
09:04:52      1043 #endif
09:04:52      1044 
09:04:52      1045     ucs_assert(uct_ib_mlx5_srq_get_wqe(&iface->rx.srq,
09:04:52 ==>  1046                                        iface->rx.srq.mask)->srq.next_wqe_index == 0);
09:04:52      1047 
09:04:52      1048     cqe = uct_rc_mlx5_iface_poll_rx_cq(iface);
09:04:52      1049     if (cqe == NULL) {
09:04:52 
09:04:52 ==== backtrace (tid:  39271) ====
09:04:52  0 0x000000000003deee uct_rc_mlx5_iface_common_poll_rx()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/0/contrib/../src/uct/ib/rc/accel/rc_mlx5.inl:1046
09:04:52  1 0x000000000062bbda uct_test::progress()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/0/contrib/../src/ucs/datastruct/callbackq.h:211
09:04:52  2 0x00000000005f8b78 uct_p2p_mix_test::random_op()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/0/contrib/../test/gtest/uct/test_p2p_mix.cc:132
09:04:52  3 0x00000000005fa130 uct_p2p_mix_test::run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/0/contrib/../test/gtest/uct/test_p2p_mix.cc:156
09:04:52  4 0x000000000055eef6 ucs::test_base::run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/0/contrib/../test/gtest/common/test.cc:276
09:04:52  5 0x0000000000549a33 testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/0/contrib/../test/gtest/common/gtest-all.cc:3562
09:04:52  6 0x000000000053dead testing::Test::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/0/contrib/../test/gtest/common/gtest-all.cc:3635
09:04:52  7 0x000000000053df7c testing::TestInfo::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/0/contrib/../test/gtest/common/gtest-all.cc:3812
09:04:52  8 0x000000000053e0df testing::TestCase::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/0/contrib/../test/gtest/common/gtest-all.cc:3930
09:04:52  9 0x0000000000542a77 testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/0/contrib/../test/gtest/common/gtest-all.cc:5802
09:04:52 10 0x0000000000542d7b testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/0/contrib/../test/gtest/common/gtest-all.cc:5719
09:04:52 11 0x00000000004ea2b3 main()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/0/contrib/../test/gtest/common/gtest.h:20059
09:04:52 12 0x0000000000021c05 __libc_start_main()  ???:0
09:04:52 13 0x0000000000528637 _start()  ???:0
09:04:52 =================================
```
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/label=hpc-test-node3,worker=0/10498/console

**Comment by yosefe** [comment]
Fixed in MOFED 4.6
